src/main/scala/strawman/collection/Iterable.scala

```diff
@@ -206,6 +206,12 @@ trait IterableOps[+A, +CC[X], +C] extends Any {
     drop(1)
   }
 
+  /** The initial part of the collection without its last element. */
+  def init: C = {
+    if (coll.isEmpty) throw new UnsupportedOperationException
+    take(size - 1)
```

Comment:
@odd Could you also introduce `dropRight` (implemented in terms of `take`), as suggested by @Ichoran?

Also, could you check that all the collection types that can provide a more efficient implementation of `dropRight` or `init` do override these operations?